packages/grid/_modules_/grid/components/cell/GridRowCells.tsx

```diff
@@ -72,32 +64,34 @@ export const GridRowCells = React.memo((props: RowCellsProps) => {
 
     const cellParams: GridCellParams = apiRef!.current.getCellParams(id, column.field);
 
-    let cssClassProp = { cssClass: '' };
-    if (column.cellClassName) {
-      if (!isFunction(column.cellClassName)) {
-        cssClassProp = { cssClass: clsx(column.cellClassName) };
-      } else {
-        cssClassProp = { cssClass: column.cellClassName(cellParams) as string };
-      }
-    }
+    const classNames = [cellClassName];
 
-    if (column.cellClassRules) {
-      const cssClass = applyCssClassRules(column.cellClassRules, cellParams);
-      cssClassProp = { cssClass: `${cssClassProp.cssClass} ${cssClass}` };
+    if (column.cellClassName) {
+      classNames.push(
+        clsx(
+          isFunction(column.cellClassName)
+            ? column.cellClassName(cellParams)
+            : column.cellClassName,
+        ),
+      );
     }
 
     const editCellState = editRowState && editRowState[column.field];
     let cellComponent: React.ReactElement | null = null;
 
     if (editCellState == null && column.renderCell) {
       cellComponent = column.renderCell(cellParams);
-      cssClassProp = { cssClass: `${cssClassProp.cssClass} MuiDataGrid-cellWithRenderer` };
+      classNames.push('MuiDataGrid-cellWithRenderer');
```

Comment:
Can we move those hardcoded class names into `cssClassesConstants.ts` and build them using the `GRID_CSS_CLASS_PREFIX`? That way if we change it in the future it will be easier to handle the change.

x-(the other places)

Reply:
Even better, we can reproduce the classes approach of the core component (dataGridClasses.ts + `generateUtilityClasses`). I have added the item to #408.